### PR TITLE
Added WSL as a new supported platform

### PR DESCRIPTION
**Pull request by lizzha**
Test command:
.\Run-LisaV2.ps1 -TestPlatform WSL -TestLocation XXX -RGIdentifier 'liztest' -XMLSecretFile XXX -TestNames "BVT-VERIFY-BOOT-ERROR-WARNINGS,PERF-GOLANG-BENCHMARK" -OsVHD 'https://aka.ms/wsl-ubuntu-1804' -DestinationOsVHDPath d:\lizzha\test

Test results:
[LISAv2 Test Results Summary]
Test Run On           : 01/30/2019 09:37:24
VHD Under Test        : https://aka.ms/wsl-ubuntu-1804
Total Test Cases      : 2 (2 Pass, 0 Fail, 0 Abort)
Total Time (dd:hh:mm) : 0:0:8

   ID TestCaseName                                                                TestResult TestDuration(in minutes)
---------------------------------------------------------------------------------------------------------------------
    1 BVT-VERIFY-BOOT-ERROR-WARNINGS                                                    PASS                  1.3
    2 PERF-GOLANG-BENCHMARK                                                             PASS                 6.21

Logs can be found at D:\LISAv2\AQ31\TestResults\2019-30-01-17-36-54-4814

### CODE REVIEW

File: LISAv2-Framework.psm1 [hunk 1]
```diff
@@ -15,7 +16,7 @@ function Start-LISAv2 {
 		[string] $ParametersFile = "",
 
 		# [Required]
-		[ValidateSet('Azure','HyperV','OL', IgnoreCase = $false)]
+		[ValidateSet('Azure','HyperV','OL','WSL',IgnoreCase = $false)]
```
[the remark this answers is not in the document]
Comment (lizzha) [reply LISAv2-Framework.psm1]: I will update the document

File: Libraries/TestLogs.psm1 [hunk 2]
```diff
@@ -180,7 +180,7 @@ Function Get-SystemBasicLogs($AllVMData, $User, $Password, $currentTestData, $Cu
 		if ($TestPlatform -ne "OL") {
```
[the remark this answers is not in the document]
Comment (lizzha) [reply Libraries/TestLogs.psm1]: yes

File: XML/GlobalConfigurations.xml [hunk 3]
```diff
@@ -53,4 +53,30 @@
             <HostFwdPort>1234</HostFwdPort>
         </TestCredentials>
     </OL>
+    <WSL>
```
[the remark this answers is not in the document]
Comment (lizzha) [reply XML/GlobalConfigurations.xml]: In WSLController, it checks whether WSL is enabled on the host, and will abort if it's not enabled

File: LISAv2-Framework.psm1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (lizzha) [reply LISAv2-Framework.psm1]: readme updated

File: XML/GlobalConfigurations.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (lizzha) [reply XML/GlobalConfigurations.xml]: comments added in the xml file, and readme updated for how to run test on WSL

**Comment by ader1990** [comment]
@lizzha can you add in the docs how to install/configure WSL on a Hyper-V node? thanks.

**Comment by lizzha** [comment]
@ader1990 added in readme

File: Libraries/TestHelpers.psm1 [hunk 4]
```diff
@@ -1061,4 +1061,43 @@ Function Remove-InvalidCharactersFromFileName
...
+				try {
+					Test-Connection -ComputerName $ipAddress.IPAddress | Out-Null
+					return $ipAddress.IPAddress
+				} catch {
+					Write-LogWarn "Fail to connect to IP $($ipAddress.IPAddress) of host $ComputerName"
+				}
+			}
+		}
+	} else {
+		$ipAddresses = Get-NetIPAddress -AddressFamily IPv4 -AddressState Preferred -CimSession $ComputerName | `
```
Comment (ader1990) [inline Libraries/TestHelpers.psm1]: here is a bit of duplicate code, but it can be refactored at a later time.

**Pull request merged**